distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1000,104 +1013,184 @@ public void testCswCreateEventEndpoint() throws Exception {
+        verifyActivity(JsonPath.from(activities.get(3)), filename, "Resource retrieval downloading . ", "RUNNING");
+        verifyActivity(JsonPath.from(activities.get(4)), filename, "Resource retrieval retrying after 1 bytes. Attempt 2 of 3.", "RUNNING");
+        verifyActivity(JsonPath.from(activities.get(5)), filename, "Resource retrieval downloading . ", "RUNNING");
+        verifyActivity(JsonPath.from(activities.get(6)), filename, "Resource retrieval completed, 8 bytes retrieved. ", "COMPLETE");
+        
+        cometDClient.shutdown();
     }
 
     /**
      * Tests that if the endpoint disconnects 3 times, the retrieval fails after 3 attempts
-     *
      * @throws Exception
      */
     @Test
     public void testRetrievalReliabilityFails() throws Exception {
+        final String cometDEndpointUrl = COMETD_ENDPOINT.getUrl();
+        final String notificationsChannel = "/ddf/notifications/**";
+        final String activitiesChannel = "/ddf/activities/**";
+        final String filename = "product3.txt";
+        
+        CometDClient cometDClient = new CometDClient(cometDEndpointUrl);
+        cometDClient.start();
+        cometDClient.subscribe(notificationsChannel);
+        cometDClient.subscribe(activitiesChannel);
         cswServer.whenHttp()
                 .match(Condition.get("/services/csw"),
                         Condition.parameter("request", "GetRecordById"))
-                .then(getCswRetrievalHeaders(),
-                        chunkedContentWithHeaders(STUB_SERVER_TEST_STRING,
-                                Duration.ofMillis(200),
-                                3));
+                .then(getCswRetrievalHeaders(filename), chunkedContentWithHeaders(STUB_SERVER_TEST_STRING, Duration.ofMillis(200), 3));
 
         String restUrl =
                 REST_PATH.getUrl() + "sources/" + CSW_STUB_SOURCE_ID + "/" + CSW_STUB_RESOURCE_ID
-                        + "?transform=resource";
+                        + "?transform=resource" + "&session=" + cometDClient.getClientId();
 
         // Verify that product retrieval fails from the csw stub server.
         // @formatter:off
         when().get(restUrl).then().log().all().assertThat().statusCode(500).contentType("text/plain")
                 .body(containsString("cannot retrieve product"));
         // @formatter:on
+
+        expect("Waiting for notifications").within(10, SECONDS).until(() -> cometDClient.getMessages(notificationsChannel).size() == 4);
+        expect("Waiting for activities").within(10, SECONDS).until(() -> cometDClient.getMessages(activitiesChannel).size() == 8);
+
+        List<String> notifications = sortJsonMessagesByTimestamp(cometDClient.getMessages(notificationsChannel));
+        assertThat(notifications.size(), is(4));
+        verifyNotification(JsonPath.from(notifications.get(0)), filename, "Resource retrieval retrying after 1 bytes. Attempt 1 of 3.", "retrying");
```

Comment:
Unfortunately that is the message that is returned in the notification.  This code is simply doing an assertion to make sure the message matches.